### PR TITLE
implemented new velocity_smoother

### PR DESCRIPTION
**Pull request by benmaidel**
shamelessly stolen and modified from [yocs](https://github.com/yujinrobot/yujin_ocs)

successfully tested on raw3-3

### CODE REVIEW

File: cob_base_velocity_smoother/CMakeLists.txt [hunk 1]
```diff
@@ -14,15 +16,30 @@ endif()
...
+  ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+  LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+  RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+)
+install(DIRECTORY param
+  DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
+)
+install(DIRECTORY cfg
+  DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
+)
```
Comment (fmessmer) [inline cob_base_velocity_smoother/CMakeLists.txt:45]: I think we usually do not install the cfg directory

File: cob_base_velocity_smoother/package.xml [hunk 2]
```diff
@@ -8,13 +8,16 @@
   <url type="website">http://ros.org/wiki/cob_base_velocity_smoother</url>
   <!-- <url type="bugtracker"></url> -->
 
-  <maintainer email="frm@ipa.fhg.de">Florian Mirus</maintainer>
+  <maintainer email="bnm@ipa.fhg.de">Benjamin Maidel</maintainer>
   <author email="frm@ipa.fhg.de">Florian Mirus</author>
```
Comment (mgruhler) [inline cob_base_velocity_smoother/package.xml:12]: a general question, just noticed it here:
how do we handle people having left IPA in terms of email addresses?
I think it is possible to remove the emails from the `author` tags, but not from the `maintainer` tags.

So, I suggest to remove the email address here, as it is not valid anymore...

File: cob_base_velocity_smoother/CMakeLists.txt [hunk 3]
```diff
@@ -14,15 +16,30 @@ endif()
...
 
 ### BUILD ###
-include_directories(include/cob_base_velocity_smoother ${catkin_INCLUDE_DIRS} ${Boost_INCLUDE_DIRS})
+include_directories(include include/cob_base_velocity_smoother ${catkin_INCLUDE_DIRS} ${Boost_INCLUDE_DIRS})
 
 add_executable(${PROJECT_NAME} src/cob_base_velocity_smoother.cpp)
 add_dependencies(${PROJECT_NAME} ${catkin_EXPORTED_TARGETS})
 target_link_libraries(${PROJECT_NAME} ${catkin_LIBRARIES} ${BOOST_LIBRARIES})
 
+add_executable(velocity_smoother src/cob_base_velocity_smoother/velocity_smoother.cpp src/velocity_smoother_node.cpp)
```
Comment (mgruhler) [inline cob_base_velocity_smoother/CMakeLists.txt:25]: any particular reason why the `velocity_smooter.cpp` is in a subdirectory in `src`?

Comment (benmaidel) [inline cob_base_velocity_smoother/CMakeLists.txt:25]: normally i put all source code which belongs to a library into one source folder.

Comment (mgruhler) [inline cob_base_velocity_smoother/CMakeLists.txt:25]: fine for me. Have just not yet seen this. Seems I never looked at a package from @ipa-bnm ;-)

File: cob_base_velocity_smoother/package.xml (hunk 2 above)
Comment (mgruhler) [inline cob_base_velocity_smoother/package.xml:12]: suggestion from @ipa-fxm was to use the mail address of his GitHub Account: https://github.com/fmirus

**Comment by floweisshardt** [comment]
FR: OK

**Comment by fmessmer** [comment]
email and install tag have been fixed in ipa320 directly

**Pull request merged**